File: train-station-api-go/pkg/domain/entities/station.go
package entities

type Station struct {
	RecordID        string    `json:"recordid" validate:"required"`
	DatasetID       string    `json:"datasetid" validate:"required"`
	IsFavorite      *bool     `json:"is_favorite" validate:"required"`
	Libelle         string    `json:"libelle" validate:"required"`
	RecordTimestamp string    `json:"record_timestamp" validate:"required"`
	Fields          *Fields   `json:"fields,omitempty" validate:"required"`
	Geometry        *Geometry `json:"geometry,omitempty" validate:"required"`
}

File: train-station-front/src/lib/entities/station.ts
import type { Fields } from '$lib/entities/fields';
import type { Geometry } from '$lib/entities/geometry';

export interface Station {
	recordid: string;
	datasetid: string;
	is_favorite: boolean;
	libelle: string;
	fields?: Fields;
	geometry?: Geometry;
	record_timestamp: string;
}
